sherpa-onnx/csrc/online-transducer-nemo-model.h

```diff
@@ -0,0 +1,148 @@
+// sherpa-onnx/csrc/online-transducer-nemo-model.h
+//
+// Copyright (c)  2024  Xiaomi Corporation
+// Copyright (c)  2024  Sangeet Sagar
+
+#ifndef SHERPA_ONNX_CSRC_ONLINE_TRANSDUCER_NEMO_MODEL_H_
+#define SHERPA_ONNX_CSRC_ONLINE_TRANSDUCER_NEMO_MODEL_H_
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#if __ANDROID_API__ >= 9
+#include "android/asset_manager.h"
+#include "android/asset_manager_jni.h"
+#endif
+
+#include "onnxruntime_cxx_api.h"  // NOLINT
+#include "sherpa-onnx/csrc/online-model-config.h"
+
+namespace sherpa_onnx {
+
+// see
+// https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/models/hybrid_rnnt_ctc_bpe_models.py#L40
+// Its decoder is stateful, not stateless.
+class OnlineTransducerNeMoModel {
+ public:
+  explicit OnlineTransducerNeMoModel(const OnlineModelConfig &config);
+
+#if __ANDROID_API__ >= 9
+  OnlineTransducerNeMoModel(AAssetManager *mgr,
+                            const OfflineModelConfig &config);
+#endif
+  
+  ~OnlineTransducerNeMoModel();
+    // A list of 3 tensors:
+  //  - cache_last_channel
+  //  - cache_last_time
+  //  - cache_last_channel_len
+  std::vector<Ort::Value> GetInitStates() const;
+
+  /** Stack a list of individual states into a batch.
+   *
+   * It is the inverse operation of `UnStackStates`.
+   *
+   * @param states states[i] contains the state for the i-th utterance.
+   * @return Return a single value representing the batched state.
+   */
+  std::vector<Ort::Value> StackStates(
+      const std::vector<std::vector<Ort::Value>> &states) const;
+
+  /** Unstack a batch state into a list of individual states.
+   *
+   * It is the inverse operation of `StackStates`.
+   *
+   * @param states A batched state.
+   * @return ans[i] contains the state for the i-th utterance.
+   */
+  std::vector<std::vector<Ort::Value>> UnStackStates(
+      const std::vector<Ort::Value> &states) const;
+
+  /** Run the encoder.
+   *
+   * @param features  A tensor of shape (N, T, C). It is changed in-place.
+   * @param states  It is from GetInitStates() or returned from this method.
+   * 
+   * @return Return a tuple containing:
+   *           - encoder_out, a tensor of shape (N, T', encoder_out_dim)
```

Comment:
Please also document `ans[1:]` in addition to `ans[0]`.

Reply:
Doc updated.